x-pack/plugins/observability/server/services/slo/summary_cleanup_task.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, Logger, SavedObjectsClient } from '@kbn/core/server';
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { SortResults } from '@elastic/elasticsearch/lib/api/types';
+import { SLO_SUMMARY_DESTINATION_INDEX_PATTERN } from '../../assets/constants';
+import { StoredSLO } from '../../domain/models';
+import { SO_SLO_TYPE } from '../../saved_objects';
+
+export const TASK_TYPE = 'SLO-DEFINITIONS-CLEANUP-TASK';
+
+function findMissingItems(soIds: string[], sloDefIds: string[]) {
+  const missingSloDefIds = sloDefIds.filter((item) => !soIds.includes(item));
+  const missingSOIds = soIds.filter((item) => !sloDefIds.includes(item));
+
+  return {
+    missingSloDefIds,
+    missingSOIds,
+  };
+}
+
+export class SloSummaryCleanupTask {
+  private abortController = new AbortController();
+  private logger: Logger;
+  private taskManager?: TaskManagerStartContract;
+  private soClient?: SavedObjectsClient;
+  private esClient?: ElasticsearchClient;
+
+  constructor(taskManager: TaskManagerSetupContract, logger: Logger) {
+    this.logger = logger;
+
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'SLO Definitions Cleanup Task',
+        timeout: '1m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              const runAt = new Date().toISOString();
+
+              if (this.soClient && this.esClient) {
+                const finder = this.soClient.createPointInTimeFinder<StoredSLO>({
+                  type: SO_SLO_TYPE,
+                  perPage: 1000,
+                  fields: ['id'],
```

Comment:
I think we need the `id` as well as the `revision`. 

For context, when we update an SLO, we bump its `revision`. The rollup and summary indices always contain the slo id and the revision. So we don't double count the data generated by the previous revision. And we can delete it asynchronously.

And one thing that can happen is that we update an SLO, we bump its revision, create the new transform (slo-id-revision) and never delete the summary document for the previous revision (for x reason). This is one way of getting an orphan SLO on the list page. The other way is when we delete an SLO, but somehow we don't delete the summary document for that slo id and revision neither.